src/main/java/past/cwdt/bookstore/dto/user/UserRegistrationRequestDto.java

```diff
@@ -0,0 +1,16 @@
+package past.cwdt.bookstore.dto.user;
+
+import jakarta.validation.constraints.Email;
+import jakarta.validation.constraints.NotBlank;
+import jakarta.validation.constraints.Size;
+import past.cwdt.bookstore.security.constraint.FieldMatch;
+
+@FieldMatch(message = "Passwords are not equal")
+public record UserRegistrationRequestDto(
+        @NotBlank @Email String email,
```

Comment:
```suggestion
        @NotBlank @Email(regexp = "your custom regex") String email,
```
by default, the email annotation is checked against the regular expression ".*". That is, any string is valid. Therefore, I would recommend adding any regular expression (because now this annotation does not validate at all)